contrib/fparser/fparser_ad.cc

```diff
@@ -333,13 +333,17 @@ typename ADImplementation<Value_t>::CodeTreeAD ADImplementation<Value_t>::D(cons
     case cFloor:
     case cCeil:
     case cTrunc:
+      throw RefuseToTakeCrazyDerivativeException;
```

Comment:
Trunc rounds towards zero. Yeah, I could do that. I guess I was trying to
retain a shred of mathematical integrity ;-). Those functions have an
infinite number of undefined points in their derivatives, while the ones
I'm allowing now only have one single one.

On Wed, Apr 29, 2015, 20:39 roystgnr <notifications@github.com> wrote:

> In contrib/fparser/fparser_ad.cc
> <https://github.com/libMesh/libmesh/pull/553#discussion_r29400224>:
>
> > @@ -333,13 +333,17 @@ typename ADImplementation<Value_t>::CodeTreeAD ADImplementation<Value_t>::D(cons
> >      case cFloor:
> >      case cCeil:
> >      case cTrunc:
> > +      throw RefuseToTakeCrazyDerivativeException;
>
> I don't know what cTrunc does, but why not allow returning 0 derivatives
> of floor() and ceil() ? They've got the same "zero everywhere except at a
> jump" structure as the comparison operators do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/libMesh/libmesh/pull/553/files#r29400224>.
>